pandas/core/frame.py

```diff
@@ -292,10 +292,17 @@
 
       .. versionadded:: 1.2.0
 
-on : label or list
+on : label or list, or Callable
     Column or index level names to join on. These must be found in both
     DataFrames. If `on` is None and not merging on indexes then this defaults
     to the intersection of the columns in both DataFrames.
+    If Callable this will perform a conditional merge. This allows you to
+    provide a custom match condition for the join. The callable takes 2
+    arguments, which represent the left and right sides of the merge, and
+    must return a boolean mask. Callable `on` is currently implemented as a
+    chunked cross-join and filtered by the callable provided. Left and right
+    chunk sizes can be set with the `get_option("conditional_merge.chunk_size")`
```

Comment:
Good to mention here that the default is 1000

Reply:
done